src/Connection.php

```diff
@@ -1070,23 +1069,32 @@ public function executeCacheQuery($sql, $params, $types, QueryCacheProfile $qcp)
         $item = $resultCache->getItem($cacheKey);
 
         if ($item->isHit()) {
-            $data = $item->get();
+            $value = $item->get();
             // is the real key part of this row pointers map or is the cache only pointing to other cache keys?
-            if (isset($data[$realKey])) {
-                $result = new ArrayResult($data[$realKey]);
-            } elseif (array_key_exists($realKey, $data)) {
+            if (isset($value[$realKey])) {
+                $result = new ArrayResult($value[$realKey]);
+            } elseif (array_key_exists($realKey, $value)) {
```

Comment:
This line isn't covered by a test and looks redundant. An empty result should be represented in the cache as an empty array, not as NULL. Even if some old caches contain NULL, a requirement to clear the cache during an upgrade doesn't look like a BC break.